editoast/src/views/scenario/macro_nodes.rs

```diff
@@ -0,0 +1,614 @@
+    pub full_name: Option<String>,
+    pub connection_time: i64,
+    pub labels: Vec<Option<String>>,
+    pub trigram: Option<String>,
+    pub path_item_key: String,
+}
+
+impl MacroNodeForm {
+    pub fn into_macro_node_changeset(self, scenario_id: i64) -> Result<Changeset<MacroNode>> {
+        let macro_node = MacroNode::changeset()
+            .scenario_id(scenario_id)
+            .position_x(self.position_x)
+            .position_y(self.position_y)
+            .full_name(self.full_name)
+            .connection_time(self.connection_time)
+            .labels(self.labels)
+            .trigram(self.trigram)
+            .path_item_key(self.path_item_key);
+
+        Ok(macro_node)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, ToSchema)]
+pub struct MacroNodeResponse {
+    pub id: i64,
+    pub position_x: f64,
+    pub position_y: f64,
+    pub full_name: Option<String>,
+    pub connection_time: i64,
+    pub labels: Vec<Option<String>>,
+    pub trigram: Option<String>,
+    pub path_item_key: String,
+}
+
+impl MacroNodeResponse {
+    pub fn new(node: MacroNode) -> Self {
+        Self {
+            id: node.id,
+            position_x: node.position_x,
+            position_y: node.position_y,
+            full_name: node.full_name,
+            connection_time: node.connection_time,
+            labels: node.labels,
+            trigram: node.trigram,
+            path_item_key: node.path_item_key,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, PartialEq, Clone, ToSchema)]
+#[cfg_attr(test, derive(Deserialize))]
+pub struct MacroNodeListResponse {
+    #[serde(flatten)]
+    stats: PaginationStats,
+    results: Vec<MacroNodeResponse>,
+}
+
+/// Get macro node list by scneario id
+#[utoipa::path(
+    get, path = "",
+    tag = "scenarios",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam, PaginationQueryParam),
+    responses(
+        (status = 200, body = MacroNodeListResponse, description = "List of macro nodes for the requested scenario"),
+        (status = 404, body = InternalError, description = "The requested scenario was not found"),
+    )
+)]
+async fn list(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Path((project_id, study_id, scenario_id)): Path<(i64, i64, i64)>,
+    Query(pagination_params): Query<PaginationQueryParam>,
+) -> Result<Json<MacroNodeListResponse>> {
+    let conn = &mut db_pool.get().await?;
+
+    // Checking role
+    let authorized = auth
+        .check_roles([BuiltinRole::OpsRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    // Check for project / study / scenario
+    check_project_study_scenario(conn, project_id, study_id, scenario_id).await?;
+
+    // Ask the db
+    let settings = pagination_params
+        .validate(1000)?
+        .warn_page_size(100)
```

Comment:
```suggestion
    let settings = pagination_params
        .validate(100)?
```

Since this is a new endpoint I suggest truly enforcing the pagination, wdyt?